core/market/decentralized/src/testing/events_helper.rs

```diff
@@ -16,11 +19,23 @@ pub mod requestor {
             _ => anyhow::bail!("Invalid event Type. ProposalEvent expected"),
         })
     }
+
+    pub async fn query_proposal(
```

Comment:
Maybe we can go even further, and define a trait for MarketService? Non-blocking comment